src/plugins/intel_cpu/src/emitters/snippets/riscv64/jit_memory_emitters.cpp

```diff
@@ -0,0 +1,317 @@
+    auto sew = (byte_size == 2) ? Xbyak_riscv::SEW::e16 : Xbyak_riscv::SEW::e32;
+    h->vsetivli(Xbyak_riscv::zero, count, sew, Xbyak_riscv::LMUL::m1);
+
+    // Load vector data from memory
+    if (compiled_byte_offset == 0) {
+        if (byte_size == 2) {
+            h->vle16_v(dst, src);
+        } else {
+            h->vle32_v(dst, src);
+        }
+    } else {
+        // Use temporary register to calculate address with offset
+        auto tmp_gpr = Xbyak_riscv::Reg(aux_gpr_idxs.empty() ? Xbyak_riscv::t0.getIdx() : aux_gpr_idxs[0]);
+        h->addi(tmp_gpr, src, static_cast<int32_t>(compiled_byte_offset));
+        if (byte_size == 2) {
+            h->vle16_v(dst, tmp_gpr);
+        } else {
+            h->vle32_v(dst, tmp_gpr);
+        }
+    }
+}
+
+void jit_load_memory_emitter::emit_data() const {
+    // No additional data emission needed for basic load
+}
+
+jit_store_memory_emitter::jit_store_memory_emitter(jit_generator_t* h, cpu_isa_t isa, const ExpressionPtr& expr)
+    : jit_memory_emitter(h, isa, expr, emitter_in_out_map::vec_to_gpr) {
+    bool is_supported_precision =
+        any_of(dst_prc, ov::element::f32, ov::element::i32, ov::element::f16) && src_prc == dst_prc;
+    OV_CPU_JIT_EMITTER_ASSERT(is_supported_precision, "Unsupported precision pair.");
+
+    const auto store = ov::as_type_ptr<snippets::op::Store>(expr->get_node());
+    OV_CPU_JIT_EMITTER_ASSERT(store != nullptr, "Expects Store expression");
+    count = store->get_count();
+    byte_size = dst_prc.size();
+}
+
+void jit_store_memory_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+    if (host_isa_ == ov::intel_cpu::riscv64::gv) {
+        emit_isa<ov::intel_cpu::riscv64::gv>(in, out);
+    } else {
+        OV_CPU_JIT_EMITTER_THROW("Doesn't support isa ", host_isa_);
+    }
+}
+
+template <cpu_isa_t isa>
+void jit_store_memory_emitter::emit_isa(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+    auto src = Xbyak_riscv::VReg(in[0]);
+    auto dst = Xbyak_riscv::Reg(out[0]);
+
+    // Set vector configuration for the store
+    auto sew = (byte_size == 2) ? Xbyak_riscv::SEW::e16 : Xbyak_riscv::SEW::e32;
+    h->vsetivli(Xbyak_riscv::zero, count, sew, Xbyak_riscv::LMUL::m1);
+
+    // Store vector data to memory
+    if (compiled_byte_offset == 0) {
+        if (byte_size == 2) {
+            h->vse16_v(src, dst);
+        } else {
+            h->vse32_v(src, dst);
+        }
+    } else {
+        // Use temporary register to calculate address with offset
+        auto tmp_gpr = Xbyak_riscv::Reg(aux_gpr_idxs.empty() ? Xbyak_riscv::t0.getIdx() : aux_gpr_idxs[0]);
+        h->addi(tmp_gpr, dst, static_cast<int32_t>(compiled_byte_offset));
+        if (byte_size == 2) {
+            h->vse16_v(src, tmp_gpr);
+        } else {
+            h->vse32_v(src, tmp_gpr);
+        }
+    }
+}
+
+void jit_store_memory_emitter::emit_data() const {
```

Comment:
Why do we need to override `emit_data` if it is empty?

Reply:
Removed